mmdeploy/codebase/mmpose/models/backbones/litehrnet.py

```diff
@@ -27,3 +27,122 @@ def cross_resolution_weighting__forward(ctx, self, x):
         for s, a in zip(x, out)
     ]
     return out
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    'mmpose.models.backbones.litehrnet.CrossResolutionWeighting.forward',
+    backend='ncnn')
+def cross_resolution_weighting__forward__ncnn(ctx, self, x):
+    """Rewrite ``forward`` for ncnn backend.
+
+    Rewrite this function to support export ``adaptive_avg_pool2d``
+    for ncnn. Ncnn has its own implement of adaptive average pooling.
+
+    Args:
+        x (list): block input.
+    """
+    from mmdeploy.codebase.mmpose.core.ops import \
+        ncnn_adaptive_avg_pool_forward
+
+    mini_size = torch.tensor([int(_) for _ in x[-1].shape[-2:]])
+    # Replace torch.nn.functional.adaptive_avg_pool2d to a dummy onnx op,
+    # which is used for ncnn implement.
+    out = [ncnn_adaptive_avg_pool_forward(s, mini_size) for s in x[:-1]] + \
+        [x[-1]]
+    out = torch.cat(out, dim=1)
+    out = self.conv1(out)
+    out = self.conv2(out)
+    out = torch.split(out, self.channels, dim=1)
+    out = [
+        s * F.interpolate(a, size=s.size()[-2:], mode='nearest')
+        for s, a in zip(x, out)
+    ]
+    return out
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    'mmpose.models.backbones.litehrnet.ShuffleUnit.forward', backend='ncnn')
+def shuffle_unit__forward__ncnn(ctx, self, x):
+    """Rewrite ``forward`` for ncnn backend.
+
+    Rewrite this function to support export ``chunk``.
```

Comment:
Can we rewrite the chunk with split? It is more general.

Reply:
> Can we rewrite the chunk with split? It is more general. If it did support, does it means that we do not need all rewriters below?

Because chunk has a parameter `axis`, which may lead `if` statement into the function, may looks ugly.

Reply:
I don't get it, could you provide an example?

Reply:
Ok, now I have tested using torch.index_select can export model correctly, so we can rewrite torch.chunk derectly.